<?xml version="1.0"?>
<launch>
  
  <arg name="joystick" default="true"/>
  <arg name="perception" default="true"/>
  <arg name="bag_file"/>
  <arg name="start" default="0"/>
  # Use simulation time fro TFs
  <param name="use_sim_time" value="true" />
  
  <node name="siar_joy" pkg="siar_driver" type="siar_teleop_joy" output="screen" if="$(arg joystick)">
    <remap from="/cmd_vel" to="/user_cmd_vel"/>
    <remap from="/joy" to="/translated_joy"/>
    <param name="max_auto_mode" value="1"/>
    <param name="max_joy_time" value="100000000"/>
    <remap from="/operation_mode" to="/om"/>
  </node>
  <node name="joy_node" pkg="joy" type="joy_node" if="$(arg joystick)" />
  
  <!-- Automatic control -->
    <node name="siar_controller" pkg="siar_controller" type="siar_controller_node" output="screen">
      <remap from="/cmd_vel_in" to="/user_cmd_vel"/>
<!--       <remap from="/cmd_vel_out" to="/cmd_vel"/>  NOTE: use cmd_vel_out to debug--> 
      <remap from="/altitude_map" to="/test_costmap"/>
      <remap from="/trajectory_marker" to="/tm"/>
      <remap from="/best_marker" to="/bm"/>
      <remap from="/operation_mode" to="/om"/>
      
      <rosparam file="$(find siar_controller)/cfg/siar_controller_fast.yaml" command="load"  /> 
      <param name="velocityset_filename" value="$(find siar_controller)/cfg/velocity_set_fast_smooth"/>
    </node>
    
    <!-- Siar costmap -->
    <include file="$(find siar_costmap)/launch/siar_costmap_bag.launch">
      <arg name="bag_file" value="$(arg bag_file)"/>
      <arg name="start" value="$(arg start)"/>
      <arg name="camera_5" value="/a"/>
      <arg name="camera_6" value="/a"/>
    </include>
  
</launch>
